Sage/carbon
# fix(text-editor): add tabIndex 0 to editor container and update types for value and onChange FE-5582

fix #5685

### Proposed behaviour


Adds `tabIndex={0}` to `draft-js` `Editor` component so that it is found in the FocusTrap's
focusable elements. Updates the `TextEditor` types for `value` and `onChange` to replace
`Record<string, unknown>` with `EditorState`

Adds cypress test to `Dialog` to catch any potential future regressions
Updates default `Dialog` story to include `TextEditor` as a child

### Current behaviour


The `Editor` component rendered as part of `TextEditor` composition has `tabIndex` `-1` so it is filtered out of focusable elements in the focus trap

### Checklist




#### QA


### Additional context



### Testing instructions


See linked sandbox and default dialog story

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
